osclients.py

```diff
@@ -47,7 +43,7 @@ class OpenStackClients(object):
     call set_keystone_version
     """
 
-    def __init__(self, auth_url=None):
+    def __init__(self, auth_url=None, modules='auto'):
```

Comment:
Because you may want to detect as soon as possible if you have all the requirements, looking at the constructor and not revising the full code search for get_*client calls.

Anyway, the default value is auto, so you don't need to specify the modules if you don't want.